packages/content/src/content.ts

```diff
@@ -1,8 +1,292 @@
+    spatialReference: {
+      wkid: wkid || 4326
+    }
+  };
+}
+
+function itemExtentToBoundary (extent: IBBox): IHubGeography {
+  if (!extent) {
+    return;
+  }
+  return {
+    // size?
+    // center?
+    geometry: createExtent(extent[0][0], extent[0][1], extent[1][0], extent[1][1])
+  };
+}
+
+function itemToContent (item: IItem, portalApiUrl: string): IHubContent {
+  const createdDate = new Date(item.created);
+  const createdDateSource = 'item.created';
+  const properties = item.properties;
+  return Object.assign({}, item, {
+    name: item.title || item.name, // should the latter only be for file types?
+    hubType: getItemHubType(item),
+    // can we strip HTML from description, and do we need to trim it to a X chars?
+    summary: item.snippet || item.description,
+    publisher: {
+      name: item.owner,
+      username: item.owner
+    },
+    permissions: { 
+      visibility: item.access,
+      permission: item.itemControl || 'view'
+    },
+    // Hub configuration metadata from item properties
+    actionLinks: properties && properties.links,
+    hubActions: properties && properties.actions,
+    metrics: properties && properties.metrics,
+    // default boundary from item.extent
+    boundary: itemExtentToBoundary(item.extent),
+    license: { name: 'Custom License', description: item.accessInformation },
+    thumbnailUrl: getItemThumbnailUrl(item, portalApiUrl),
+    // dates and sources
+    createdDate,
+    createdDateSource,
+    publishedDate: createdDate,
+    publishedDateSource: createdDateSource,
+    updatedDate: new Date(item.modified),
+    updatedDateSource: 'item.modified'
+  });
+}
+
+/**
+ * Convert a Hub v3 Dataset to Hub Content 
+ * TODO: Change to use mdJSON translation for configurable metadata
+ *
+ * @param {IModel} item Hub Item
+ * @returns {IHubContent} Hub content object
+ * @export
+ */
+function datasetToContent(
+  // TODO: IDataset
+  dataset: any,
+  portalApiUrl: string
+): IHubContent {
+  if (!dataset) {
+    return;
+  }
+
+  // extract item from dataset and create content
+  const item = datasetToItem(dataset);
+  const content = itemToContent(item, portalApiUrl);
+
+  // overwrite or add enrichments from Hub API
+  const attributes = dataset.attributes;
+  if (!attributes) {
+    return content;
+  }
+  const {
+    // common enrichments
+    boundary,
+    modifiedProvenance,
+    // dataset enrichments
+    recordCount
+    // TODO: server, etc
+  } = attributes;
+  content.boundary = boundary && boundary;
+  content.updatedDateSource = modifiedProvenance && modifiedProvenance;
+  // type-specific enrichments
+  // TODO: should we return a different subtype of IHubContent for this?
+  // TODO: should this be based on existence of attributes instead of hubType?
+  if (content.hubType === 'dataset') {
+    content.recordCount = recordCount;
+  }  
+  // TODO: any remaining enrichments
+  return content;
+}
+
+function datasetToItem (dataset:any):IItem {
```

Comment:
use a better type than any. any allows string, numbers, etc.

Reply:
that was just temporary, I've added `DatasetResource` using https://github.com/mike-north/jsonapi-typescript